### ISSUE TITLE
模型推理部分疑问

### ISSUE THREAD

**Issue by linuxd234** [zju3dv/snake#221]
作者您好，非常感谢您能提供这么优秀的算法，在研究代码的过程中有以下几个疑问：

1.在模型推理阶段使用detection生成的bbox时：
在产生轮廓使用的centernet生成的bbox尺寸是4x下采样后的吗,或者说centernet的输出的bbox就是4x下采样后的?
lib/utils/snake/snake_gcn_utils.py line:29
（看代码prepare_testing_init-->snake_decode.get_init(box)好像没有对轮廓尺寸进行尺寸上的变换）

2.在evaluate部分，如果只评估bbox时：
lib/evaluators/coco/snake.py  对centernet的输出进行 line:107: box=detection[:,:4].detach().cpu().numpy()*snake_config.down_ratio
操作后不是已经将bbox扩大到原图级别的了吗？
为什么后边还进行了一步 line:125
box_=data_utils.affine_transform(box[i].reshape(-1,2),trans_output_inv).ravel()
对bbox进行仿射变化呢？

3.网络的输出的轮廓点以及极端预测点是原图尺寸上的吗？还是在4x下采样后的特征图上的。

希望占用您宝贵的时间为我解答一下，谢谢。

**Comment by pengsida** [comment]
1. 是的。
2. 因为在dataset里面对图片做了仿射变换。
3. 网络输出的轮廓点是在4x下采样后的特征图上的，要乘以4倍才到原图。

**Comment by linuxd234** [comment]
> 1. 是的。
> 2. 因为在dataset里面对图片做了仿射变换。
> 3. 网络输出的轮廓点是在4x下采样后的特征图上的，要乘以4倍才到原图。

谢谢大佬解答！
